data/src/main/java/com/microsoft/azure/kusto/data/auth/CloudInfo.java

```diff
@@ -127,8 +127,10 @@ public static CloudInfo retrieveCloudInfoForCluster(String clusterUrl,
                         ((Closeable) localHttpClient).close();
                     }
                 }
-            } catch (IOException | URISyntaxException ex) {
-                throw new DataServiceException(clusterUrl, "IOError when trying to retrieve CloudInfo", ex, true);
+            }catch (URISyntaxException e) {
+                throw new DataServiceException(clusterUrl, "URISyntaxException when trying to retrieve CloudInfo:" + e.getMessage(), e, true);
+            } catch (IOException ex) {
+                throw new DataServiceException(clusterUrl, "IOException when trying to retrieve CloudInfo:" + ex.getMessage(), ex, true);
```

Comment:
Why is IOExeception permanent here but not before?

Reply:
wierd - not pushed right